### PR TITLE
Migrate cut

### PR DESCRIPTION
**Pull request by namse**
# What I did 
- Migrate rpc's data type `Cut`.
- Separate namui and namui-type to use struct in non-wasm, non-game environment(in this PR, for luda-editor/server).
- Remove `Time::now()` to separate namui-type from namui. Instead, call `namui::now()`.

# Comment
I made a migration tool, this can migrate struct using secret '_v' field in json.

Currently it's not possible to load sequence in master branch because the data structure is not same, unlike what I said in https://github.com/NamseEnt/namseent/pull/410. It makes error even user didn't use image.

So I made this migration system urgently.

### CODE REVIEW

File: luda-editor/migration/src/lib.rs [hunk 1]
```diff
@@ -0,0 +1,247 @@
...
+                                while let Some((key, value)) = tri!(visitor.next_entry()) {
+                                    values.insert(key, value);
+                                }
+
+                                Ok(Value::Object(values))
+                            }
+                            None => Ok(Value::Object(Map::new())),
+                        }
+                    }
+                }
```
Comment (namse) [inline luda-editor/migration/src/lib.rs:210]: This code is from serde_json, to deserialize json using `deserializer.deserialize_any`.

File: luda-editor/migration/src/lib.rs [hunk 2]
```diff
@@ -0,0 +1,247 @@
...
+                if version < 1 {
+                    let previous = serde_json::from_value(json_value).map_err(|e| {
+                        serde::de::Error::custom(format!("failed to deserialize previous version: {}", e))
+                    })?;
+                    Ok(#struct_name::migrate(previous))
+                } else {
+                    Ok(#struct_name {
+                        #(#fields_deserialize)*
+                    })
+                }
```
Comment (bigfoodK) [inline luda-editor/migration/src/lib.rs:235]: Could you check this?
Is it right to use `#struct_name::migrate(previous)` when serialized version is lower than **1**?. not **current version**?
I expected the following logic.
```
if `serialized_version` is lower than `current_version`:
    do migrate
else:
    return deserialized
```

**Review by bigfoodK — CHANGES_REQUESTED** [review]

File: luda-editor/migration/src/lib.rs (hunk 2 above)
Comment (namse) [reply luda-editor/migration/src/lib.rs:235]: Oh my typo. give me sec

Comment (namse) [reply luda-editor/migration/src/lib.rs]: Fixed https://github.com/NamseEnt/namseent/pull/412/commits/8e3cb246ed1cf1f8652a42272e56aed93400d255

**Pull request merged**